query/src/main/java/org/infinispan/query/dsl/embedded/impl/BaseJPAFilterIndexingServiceProvider.java

```diff
@@ -199,9 +215,11 @@ void unregister() {
 
       @Override
       public void onFilterResult(Object userContext, Object eventType, Object instance, Object[] projection, Comparable[] sortProjection) {
-         CacheEntryEvent<K, V> event = (CacheEntryEvent<K, V>) userContext;
-         if (event.isPre() && isClustered || isPrimaryOnly &&
-               !clusteringDependentLogic.getCacheTopology().getDistribution(event.getKey()).isPrimary()) {
+         EventWrapper eventWrapper = (EventWrapper) userContext;
+         Object storedKey = eventWrapper.getKey();
+         CacheEntryEvent<K, V> event = (CacheEntryEvent<K, V>) eventWrapper.getEvent();
+         boolean primaryOwner = clusteringDependentLogic.getCacheTopology().getDistribution(storedKey).isPrimary();
```

Comment:
The logic is identical, but on master it could benefit from logical shortcircuiting, potentially saving the cost of ```getDistribution(k)```. now it cannot.